boilerplate/app/navigators/navigation-utilities.tsx

```diff
@@ -1,37 +1,35 @@
-import { useState, useEffect, useRef } from "react"
+import React, { useState, useEffect, useRef } from "react"
 import { BackHandler } from "react-native"
 import {
-  PartialState,
-  NavigationState,
-  NavigationAction,
   createNavigationContainerRef,
+  NavigationContainerRef,
+  Route,
 } from "@react-navigation/native"
 
-/* eslint-disable */
-export const RootNavigation = {
-  navigate(_name: string, _params?: any) {},
-  goBack() {},
-  resetRoot(_state?: PartialState<NavigationState> | NavigationState) {},
-  getRootState(): NavigationState {
-    return {} as any
-  },
-  dispatch(_action: NavigationAction) {},
-}
-/* eslint-enable */
-
 export const navigationRef = createNavigationContainerRef()
 
 /**
- * Gets the current screen from any navigation state.
+ *
+ * Gets the current route from any navigation state.
+ * returns an array with [route.key || route.name, route]
+ * route.name is a string, route is the entire route object.
+ * #### Example
+ * ```js
+ * const [currentRouteName, route] = getRoute(navigationRef)
```

Comment:
We potentially might want to make this an object destructure instead of array destructure. Array allows the developer to specify different names, but object would enforce that they're getting what they think they're getting. I think an object destructure might work better here. What do you think, @harrisrobin?

```suggestion
 * const {currentRouteName, route} = getRoute(navigationRef)
```